feat: Collapsible Panel Component (#3025)
BrightspaceUI/core: components/collapsible-panel/README.md

| `expand-collapse-label` | String | Optional label describing the contents of the header (used by screen readers) |
| `heading-style` | Number | The heading style to use |
| `heading-level` | Number | Semantic heading level (h1-h4) |
| `no-sticky` | Boolean | Disables sticky positioning for the header |
| `padding` | String | Optionally set horizontal padding of inline panels |
| `panel-title` | String | The title of the panel |
| `type` | String | The type of collapsible panel |
<!-- docs: end hidden content -->

### Panel Types

<!-- docs: demo live name:d2l-collapsible-panel size:large -->
```html
<script type="module">
	import '@brightspace-ui/core/components/collapsible-panel/collapsible-panel.js';
</script>

<d2l-collapsible-panel panel-title="Default panel">
	This is the default collapsible panel style
</d2l-collapsible-panel>

<d2l-collapsible-panel type="subtle" panel-title="Subtle panel">
	This is the "subtle" collapsible panel style
</d2l-collapsible-panel>

<d2l-collapsible-panel type="inline" panel-title="Inline panel">
	This is the "inline" collapsible panel style
</d2l-collapsible-panel>
```

## Summary Items [d2l-collapsible-panel-summary-item]

> The summary area takes information from the expanded panel and summarizes it for the collapsed version.This can help the user understand what information is inside the panel without having to click on it.

Reviewer: ```suggestion
The summary area takes information from the expanded panel and summarizes it for the collapsed version. This can help the user understand what information is inside the panel without having to click on it.
```

	<div slot="header" style="align-items: center; display: flex; gap: 0.6rem;">
		<d2l-status-indicator state="none" text="Due Today"></d2l-status-indicator>
		<p class="d2l-body-small">Posts: 1 thread, 1 reply</p>
		<d2l-link small href="https://www.d2l.com" target="blank">Link</d2l-link>
	</div>
	<div slot="summary">
		<d2l-collapsible-panel-summary-item text="Always available"></d2l-collapsible-panel-summary-item>
	</div>
	Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas odio ligula, aliquam efficitur sollicitudin non, dignissim quis nisl. Nullam rutrum, lectus sed finibus consectetur, dolor leo blandit lorem, vitae consectetur arcu enim ornare tortor. Praesent lobortis libero in libero sagittis consectetur. Maecenas ut velit efficitur, consectetur augue vitae, finibus turpis. In id tempor quam. Integer sed facilisis mi. Interdum et malesuada fames ac ante ipsum primis in faucibus. Ut a volutpat lacus. Suspendisse potenti. Quisque egestas erat urna, et accumsan est accumsan sit amet. Sed luctus vestibulum lacus. Mauris nisi orci, rhoncus sed est sit amet, pretium facilisis felis.
</d2l-collapsible-panel>
```

## Accessibility

### Panel label

> By default, the panel is described by screen readers with the `panel-title` attribute. There may be situations where the screen reader should read additional information. In this case, the aria label can be explicitly defined using the `expand-collapse-label` property.

Reviewer: Just update wording slightly to not leak implementation details. 

```suggestion
By default, the panel is described by screen readers with the `panel-title` attribute. There may be situations where the screen reader should read additional information. In this case, a special label can be specified using the `expand-collapse-label` property.
```

* Always use this in the right hand panel FACE Activity Creation experience (FACE, First Use Activity Create & Edit)
<!-- docs: end dos -->

<!-- docs: start donts -->
* Don't nest collapsible panels within each other
* Avoid using a collapsible panel for required options in a form
* Avoid long lists of collapsible panels; consider using the nested list in this case (exceptions may apply)
* Don't have interactions or elements in the closed state that are different or disappear in the open state (exception: the summary)
	* For example, don't have a button that's only available in the collapsed state of the panel
<!-- docs: end donts -->
<!-- docs: end best practices -->

## Composition

## Collapsible Panel [d2l-collapsible-panel]

The `d2l-collapsible-panel` element is a container that provides specific layout slots such as `header`, `summary`, `actions`, and a default slot for the expanded content.

<!-- docs: demo live name:d2l-collapsible-panel size:large -->
```html
<script type="module">
	import '@brightspace-ui/core/components/button/button-subtle.js';
  import '@brightspace-ui/core/components/collapsible-panel/collapsible-panel.js';

>   import '@brightspace-ui/core/components/collapsible-panel/collapsible-panel-summary-item.js';

Reviewer 2: Nit: indentation

Author: Yup 😄 demo file is still somewhat volatile

	import '@brightspace-ui/core/components/inputs/input-search.js';
	import '@brightspace-ui/core/components/filter/filter.js';
	import '@brightspace-ui/core/components/filter/filter-dimension-set.js';
</script>

> <d2l-collapsible-panel title="Availability Dates and Conditions">

Reviewer 2: ```suggestion
<d2l-collapsible-panel panel-title="Availability Dates and Conditions">
```

Author: gah...thought I caught em all.

		</div>
	</div>
</d2l-collapsible-panel>
```

<!-- docs: start hidden content -->
### Slots

| Slot | Type | Description |
|--|--|--|

| `header` | optional | Supporting header content |
| `actions` | optional | Buttons and dropdown openers to be placed in top right corner of header |
| `summary` | optional | Summary of the expanded content. Only accepts `d2l-collapsible-panel-summary-item` |
| `default` | required | Content that is rendered when the panel is expanded |

### Properties:

| Property | Type | Description |
|--|--|--|
| `expanded` | Boolean | Whether or not the panel is expanded |
| `expand-collapse-label` | String | Label describing the contents of the header (used by screen readers) |
| `full-width` | Boolean | Whether or not the content should extend to the full width (only valid when using "inline" panel type) |
| `heading-style` | Number | The heading style to use |
| `heading-level` | Number | Semantic heading level (h1-h4) |

> | `title` | String | The title of the panel |

Reviewer 2: ```suggestion
| `panel-title` | String | The title of the panel |
```
